File: Code/CustomerAgent/src/server/display_names.py
"""Display-name overrides for service names returned to the UI.

Some raw service names emitted by Log Analytics (`AppTraces.Properties.ServiceName`)
have long, internal-sounding labels (e.g., "SQL Connectivity") that are not the
shorter, customer-facing names we want surfaced in the React UI. This module
centralises the mapping so every API endpoint that returns a service name can
apply the same override.

Keep this map small and authoritative. Synonyms / aliases do NOT belong here —
this is the canonical "what should the UI show?" source. Synonyms used during
entity normalization live in
`Code/RATIO_MCP/src/datasets/ServiceNameSynonyms.json`.
"""
from __future__ import annotations

# Map of raw upstream value -> display value to surface in the UI.
SERVICE_NAME_DISPLAY_OVERRIDES: dict[str, str] = {
    "SQL Connectivity": "SQL",
}


def display_service_name(name: str | None) -> str:
    """Return the UI display name for a raw upstream service name.

    Unknown names are returned unchanged (after stripping). Empty / None
    inputs return "".
    """
    if not name:
        return ""
    stripped = name.strip()
    return SERVICE_NAME_DISPLAY_OVERRIDES.get(stripped, stripped)

File: Code/CustomerAgent/ratio_ui_web/src/api/displayNames.ts
/**
 * Display-name overrides for service names surfaced in the React UI.
 *
 * Some raw service names returned by upstream services (Log Analytics
 * `AppTraces.Properties.ServiceName`, the cloud Container App's
 * `/api/run/services`, etc.) have long, internal-sounding labels that we
 * don't want users to see. This module is the single client-side source
 * of truth for "what should we render?" overrides and is applied at the
 * API boundary so every consumer (tabs, panel headers, channel names,
 * email subjects derived from these values) gets the normalized form.
 *
 * Keep this map small and authoritative and in sync with the server-side
 * map at `Code/CustomerAgent/src/server/display_names.py`. Synonyms /
 * aliases for entity normalization belong in
 * `Code/RATIO_MCP/src/datasets/ServiceNameSynonyms.json`, NOT here.
 */
const SERVICE_NAME_DISPLAY_OVERRIDES: Record<string, string> = {
  'SQL Connectivity': 'SQL',
};

export function displayServiceName(name: string | null | undefined): string {
  if (!name) return '';
  const stripped = name.trim();
  return SERVICE_NAME_DISPLAY_OVERRIDES[stripped] ?? stripped;
}
